src/NLog/Logger.cs

```diff
@@ -128,16 +138,15 @@ public void SetProperty(string propertyKey, object propertyValue)
             if (string.IsNullOrEmpty(propertyKey))
                 throw new ArgumentException(nameof(propertyKey));
 
-            _contextProperties = CopyOnWrite(propertyKey, propertyValue);
+            _contextProperties = CreateContextPropertiesDictionary(_contextProperties);
+            _contextProperties[propertyKey] = propertyValue;
```

Comment:
This is not good. As you are modifying the active dictionary.

Reply:
not really, CreateContextPropertiesDictionary made a new reference. But anyway, I see this PR is already superseded 👼